panic at github.com/anacrolix/missinggo/copy.go:28

I tried putting http://old-releases.ubuntu.com/releases/14.04.2/ubuntu-14.04.2-desktop-amd64.iso.torrent into the torrents dir and starting torrentfs:

$ torrentfs -mountDir mnt -torrentPath tr
panic: reflect: call of reflect.Value.Len on func Value

goroutine 15 [running]:
panic(0xabe0a0, 0xc82016aa80)
	/home/ctn/go/src/runtime/panic.go:481 +0x3e6
reflect.Value.Len(0x9e2e80, 0xc82016c800, 0x13, 0x14)
	/home/ctn/go/src/reflect/value.go:1015 +0xd2
github.com/anacrolix/missinggo.CopyExact(0x9c9560, 0xc82016aa60, 0x9e2e80, 0xc82016c800)
	/home/ctn/src/github.com/anacrolix/missinggo/copy.go:28 +0x503
github.com/anacrolix/torrent/util/dirwatch.torrentFileInfoHash(0xc82016e7e0, 0x2b, 0x0, 0x0, 0x0)
	/home/ctn/src/github.com/anacrolix/torrent/util/dirwatch/dirwatch.go:72 +0x15b
github.com/anacrolix/torrent/util/dirwatch.scanDir(0x7fffffffed55, 0x2, 0xc82016e780)
	/home/ctn/src/github.com/anacrolix/torrent/util/dirwatch/dirwatch.go:103 +0xa82
github.com/anacrolix/torrent/util/dirwatch.(*Instance).refresh(0xc82010ef90)
	/home/ctn/src/github.com/anacrolix/torrent/util/dirwatch/dirwatch.go:170 +0x4d
github.com/anacrolix/torrent/util/dirwatch.New.func1(0xc820024110)
	/home/ctn/src/github.com/anacrolix/torrent/util/dirwatch/dirwatch.go:208 +0x24
created by github.com/anacrolix/torrent/util/dirwatch.New
	/home/ctn/src/github.com/anacrolix/torrent/util/dirwatch/dirwatch.go:211 +0x1f8

--- a/cmd/torrentfs/main.go
+++ b/cmd/torrentfs/main.go
@@ -137,12 +137,15 @@ func main() {
 	resolveTestPeerAddr()
 	fs := torrentfs.New(client)
 	go exitSignalHandlers(fs)
-	go func() {
-		for {
-			addTestPeer(client)
-			time.Sleep(10 * time.Second)
-		}
-	}()
+
+	if testPeerAddr != nil {
+		go func() {
+			for {
+				addTestPeer(client)
+				time.Sleep(10 * time.Second)
+			}
+		}()
+	}
 
 	if err := fusefs.Serve(conn, fs); err != nil {
 		log.Fatal(err)

--- a/util/dirwatch/dirwatch.go
+++ b/util/dirwatch/dirwatch.go
@@ -69,7 +69,7 @@ func torrentFileInfoHash(fileName string) (ih metainfo.Hash, ok bool) {
 	if mi == nil {
 		return
 	}
-	missinggo.CopyExact(ih[:], mi.Info.Hash)
+	ih = mi.Info.Hash()
 	ok = true
 	return
 }

